packages/camera/camera_android_camerax/lib/src/android_camera_camerax.dart

```diff
@@ -319,17 +354,102 @@ class AndroidCameraCameraX extends CameraPlatform {
         .bindToLifecycle(cameraSelector!, <UseCase>[preview!]);
   }
 
-  /// Unbinds [preview] instance to camera lifecycle controlled by the
-  /// [processCameraProvider].
-  Future<void> _unbindPreviewFromLifecycle() async {
-    final bool previewIsBound = await processCameraProvider!.isBound(preview!);
-    if (preview == null || !previewIsBound) {
+  /// Configures the [imageAnalysis] instance for image streaming and binds it
+  /// to camera lifecycle controlled by the [processCameraProvider].
+  Future<void> _configureAndBindImageAnalysisToLifecycle() async {
+    assert(processCameraProvider != null);
+    assert(cameraSelector != null);
+
+    if (imageAnalysis != null &&
+        await processCameraProvider!.isBound(imageAnalysis!)) {
+      // imageAnalysis already configured and bound to lifecycle.
       return;
     }
 
+    // Create Analyzer that can read image data for image streaming.
+    Future<void> analyze(ImageProxy imageProxy) async {
+      final List<PlaneProxy> planes = await imageProxy.getPlanes();
+      final List<CameraImagePlane> cameraImagePlanes = <CameraImagePlane>[];
+      for (final PlaneProxy plane in planes) {
+        cameraImagePlanes.add(CameraImagePlane(
```

Comment:
do you want to timeout and handle an error if one of these calls gets stuck? 
Also I think with the current implementation you will await the buffer then the rowStride then the pixel stride You may want to modify this code to the following so that it can kick off all 3 async calls at the same time. 

```
Future<int> bytes = plane.getBuffer();
Future<int> bytesPerRow = plane.getRowStride();
Future<int> bytesPerPixel = plane.getPixelStride();
cameraImagePlanes.add(CameraImagePlane(
            bytes: await bytes,
            bytesPerRow: await bytesPerRow,
            bytesPerPixel: await bytesPerPixel));
```